packages/migrate/test/migrate.test.ts

```diff
@@ -0,0 +1,430 @@
+import { readFileSync } from 'node:fs';
+import path, { dirname, resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { Project } from 'fixturify-project';
+import { Reporter } from '@rehearsal/reporter';
+import { afterEach, beforeEach, describe, expect, test } from 'vitest';
+import { Logger, createLogger, format, transports } from 'winston';
+import { MigrateInput, migrate } from '../src/migrate.js';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const projectPath = resolve(__dirname, 'fixtures', 'project');
+
+function changeExtension(file: string, ext: string): string {
+  const parts = path.parse(file);
+
+  ext = ext.startsWith('.') ? ext : `.${ext}`;
+
+  return path.format({
+    ...parts,
+    ext,
+    base: undefined,
+  });
+}
+
+const extLookup = {
+  '.js': '.ts',
+  '.gjs': '.gts',
+  '.hbs': '.hbs',
+  '.ts': '.ts',
+};
+
+type ValidExtension = keyof typeof extLookup;
+
+const validExtensions = Object.keys(extLookup) as Array<ValidExtension>;
+
+function isValidExtension(ext: string): ext is ValidExtension {
+  return validExtensions.includes(ext);
+}
+
+function expectFile(filePath: string): Vi.Assertion<string> {
+  return expect(readFileSync(filePath, 'utf-8'));
+}
+
+function prepareInputFiles(project: Project, files: string[] = ['index.js']): string[][] {
+  const inputs = files.map((file) => {
+    return resolve(project.baseDir, 'src', file);
+  });
+
+  const outputs = inputs.map((file) => {
+    const fileExt = path.extname(file);
+
+    if (!isValidExtension(fileExt)) {
+      throw new Error(`found a file extension we don't know about: ${fileExt}`);
+    }
+
+    const ext = extLookup[fileExt];
+
+    return changeExtension(file, ext);
+  });
+
+  return [inputs, outputs];
+}
+
+describe('migrate', () => {
+  describe('.gjs', () => {
+    let project: Project;
+    let logger: Logger;
+    let reporter: Reporter;
+
+    beforeEach(async () => {
+      project = Project.fromDir(projectPath, { linkDeps: true, linkDevDeps: true });
+
+      delete project.files['index.js'];
+
+      await project.write();
+
+      logger = createLogger({
+        transports: [new transports.Console({ format: format.cli(), level: 'debug' })],
+      });
+
+      reporter = new Reporter(
+        {
+          tsVersion: '',
+          projectName: '@rehearsal/test',
+          basePath: project.baseDir,
+          commandName: '@rehearsal/migrate',
+        },
+        logger
+      );
+    });
+
+    afterEach(() => {
+      project.dispose();
+    });
+
+    test('with bare template', async () => {
+      const [inputs, outputs] = prepareInputFiles(project, ['template-only.gjs']);
+
+      const input: MigrateInput = {
+        basePath: project.baseDir,
+        sourceFiles: inputs,
+        entrypoint: '',
+        reporter,
+        logger,
+      };
+
+      await migrate(input);
+
+      const expected = `<template>
```

Comment:
Is there a reason why we want this here instead of using snapshots?

Reply:
There are a lot of these. @cafreeman if snapshots aren't ideal; what if we just move these to static .output test files?

Reply:
@lynchbomb I'm not opposed to moving to snapshot tests if we really want to do that, but I honestly think having the inlined expected outputs is nice since it makes it *very* easy to see what `migrate` is actually doing.